### DISCUSSION TITLE
Added: LDM notes for June 22, 2020

### DISCUSSION THREAD

**Discussion by agocke** [dotnet/csharplang#3623] [General]
## Jun 22, 2020

[C# Language Design Notes for June 22, 2020](https://github.com/dotnet/csharplang/blob/master/meetings/2020/LDM-2020-06-22.md)

1. Data properties

1. Clarifying what's supported in records for C# 9

    - Structs

    - Inheritance with records and classes
 
    - `with` expressions on non-records

**Comment by HaloFour** [comment]
The decision around the syntax for declaring a record was made under the assumption that structs could probably be automatically considered records due to their existing value equality properties.  Now that the gap between records and structs widens due to interface implementations and potentially other members to the extent that the team is leaning towards explicit opt-in via "struct records" where does this leave the previous syntax decision?

**Comment by CyrusNajmabadi** [comment]
I don't see it really changing things. 'record' is the nice short way to make a normal record. 'struct record' is just a number way to say: I really the same, but it should be a value type

**Comment by HaloFour** [comment]
So rather than `record` always being a modifier for either reference or value types, we want `struct` to sometimes be a type and sometimes be a modifier to some third type that is normally a reference type?

This is the approach that Java is taking, but Java's approach makes significantly more sense since `inline` will be a modifier to both `class` and `record`.  We don't have 3.5 different "type"-like things in play.

**Comment by CyrusNajmabadi** [comment]
> So rather than record always being a modifier for either reference or value types, we want struct to sometimes be a type and sometimes be a modifier to some third type that is normally a reference type?

Currently yes.  For people who want value-type records, you use two keywords in both events.  However for lots of people where they only want/nee ref-type records, you only need a single one.  That's what we opted for as being preferable overall.

**Comment by ufcpp** [comment]
> For people who want value-type records, you use two keywords in both events.

If so, I think I will be using always record structs and recommend for my community members not to use old-fashioned structs anymore except  for compatibility reason.

**Comment by CyrusNajmabadi** [comment]
Ok.
